lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -202,36 +206,29 @@ def as_hda(obj_dct):
                     self.sa_session.add(imported_job)
                     self.sa_session.flush()
 
-                    class HistoryDatasetAssociationIDEncoder(json.JSONEncoder):
-                        """ Custom JSONEncoder for a HistoryDatasetAssociation that encodes an HDA as its ID. """
+                    def remap_objects(p, k, obj):
+                        if isinstance(obj, dict) and obj.get('__HistoryDatasetAssociation__', False):
+                            return (k, hdas_by_key[obj[object_key]].id)
+                        return (k, obj)
 
-                        def default(self, obj):
-                            """ Encode an HDA, default encoding for everything else. """
-                            if isinstance(obj, model.HistoryDatasetAssociation):
-                                return obj.id
-                            return json.JSONEncoder.default(self, obj)
+                    from boltons.iterutils import remap
```

Comment:
This should be imported at the top, right ?

Reply:
Yup - I'll rebase with a fix for that and resolve this.